src/Cake.Common/Solution/SolutionParser.cs

```diff
@@ -182,5 +195,104 @@ private static void ParseNestedProjectLine(List<SolutionProject> projects, strin
+        {
+            var folderName = xmlElement.GetAttribute("Name");
+            if (string.IsNullOrWhiteSpace(folderName))
+            {
+                throw new CakeException($"Could not find solution folder name attribute in solution file '{solutionFile.Path.FullPath}'.");
+            }
+
+            // the name of the folder is also its path. And it always has a starting and trailing "/", which break the check if its relative, so they are removed.
+            var folderPath = new FilePath(folderName.Trim('/'));
+            var folderFullPath = folderPath.IsRelative ?
+                solutionFile.Path.GetDirectory().CombineWithFilePath(folderPath) :
+                folderPath;
+
+            var actualFolderName = folderName.Split('/', StringSplitOptions.RemoveEmptyEntries).Last();
+
+            // elements in the SLNX file don't have any IDs anymore, so it's omitted.
+            var solutionFolder = new SolutionFolder(string.Empty, actualFolderName, folderFullPath);
+            var projects = new List<SolutionProject>
+            {
+                solutionFolder,
+            };
+
+            foreach (var childElement in xmlElement.ChildNodes.OfType<XmlElement>())
+            {
+                var childProjects = ParseSlnxElement(childElement, solutionFile);
+                foreach (var childProject in childProjects)
+                {
+                    childProject.Parent = solutionFolder;
+                    solutionFolder.Items.Add(childProject);
+                    projects.Add(childProject);
+                }
+            }
+
+            return projects;
+        }
+
+        private static List<SolutionProject> ParseSlnxProject(XmlElement xmlElement, IFile solutionFile)
+        {
+            const string defaultProjectTypeId = "{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}";
+
+            var projectPath = xmlElement.GetAttribute("Path");
+            if (string.IsNullOrWhiteSpace(projectPath))
+            {
+                throw new CakeException($"Could not find solution project path attribute in solution file '{solutionFile.Path.FullPath}'.");
+            }
+
+            var projectTypeId = xmlElement.GetAttribute("Type");
+            projectTypeId = string.IsNullOrWhiteSpace(projectTypeId)
+                ? defaultProjectTypeId
+                // the new project type id notation does not quite fit the known format, so it is adjusted.
+                : $"{{{projectTypeId.ToUpper()}}}";
+
+            var projectFilePath = new FilePath(projectPath);
+            var projectFileFullPath = projectFilePath.IsRelative ?
+                solutionFile.Path.GetDirectory().CombineWithFilePath(projectFilePath) :
+                projectFilePath;
+
+            // the project name is part of its path.
+            var actualProjectName = projectPath.Split('/', StringSplitOptions.RemoveEmptyEntries).Last().Replace(".csproj", string.Empty);
```

Comment:
We could already created file path here.
```suggestion
            var actualProjectName = projectFilePath.GetFilenameWithoutExtension().FullPath;
```

Reply:
Addressed this with my latest changes.